### PR TITLE
Improve performance twenty orm

### PR DESCRIPTION
**Pull request by charlesBochet**
## Context

As we grow, the messaging scripts are experiencing performance issues forcing us to temporarily disable them on the cloud.
While investigating the performance, I have noticed that generating the entity schema (for twentyORM) in the repository is taking ~500ms locally on my Mac M2 so likely more on pods. Caching the entitySchema then!

I'm also clarifying naming around schemaVersion and cacheVersions ==> both are renamed workspaceMetadataVersion and migrated to the workspace table (the workspaceCacheVersion table is dropped).

### CODE REVIEW

File: packages/twenty-server/src/engine/core-modules/file/controllers/file.controller.ts [hunk 1]
```diff
@@ -44,6 +44,10 @@ export class FileController {
         workspaceId,
       );
 
+      fileStream.on('error', () => {
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/core-modules/file/controllers/file.controller.ts:47]: I have noticed that local driver (at least) does not throw FileNotFound as the stream get piped. do we  have to change the API here of our file drivers or is there a way to make it work like the s3 driver?
@Weiko any idea?

File: packages/twenty-server/src/engine/twenty-orm/factories/workspace-datasource.factory.ts [hunk 2]
```diff
@@ -78,18 +78,43 @@ export class WorkspaceDatasourceFactory {
           );
 
         if (!dataSourceMetadata) {
-          throw new Error('Data source metadata not found');
+          throw new Error(
+            `Data source metadata not found for workspace ${workspaceId}`,
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/twenty-orm/factories/workspace-datasource.factory.ts:82]: adding a bit more logs to ease debugging

File: packages/twenty-server/src/engine/twenty-orm/factories/workspace-datasource.factory.ts [hunk 3]
```diff
@@ -78,18 +78,43 @@ export class WorkspaceDatasourceFactory {
...
         }
 
         if (!cachedObjectMetadataCollection) {
-          throw new Error('Object metadata collection not found');
+          throw new Error(
+            `Object metadata collection not found for workspace ${workspaceId}`,
+          );
+        }
+
+        const cachedEntitySchemaOptions =
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/twenty-orm/factories/workspace-datasource.factory.ts:92]: we are caching the entity schema options (this is the typeOrmNaming, not sure it's super clear)

File: packages/twenty-server/src/modules/messaging/message-import-manager/crons/jobs/messaging-messages-import.cron.job.ts [hunk 4]
```diff
@@ -69,5 +70,7 @@ export class MessagingMessagesImportCronJob {
         }
       }
     }
+
+    console.timeEnd('MessagingMessagesImportCronJob time');
```
Comment (charlesBochet) [inline packages/twenty-server/src/modules/messaging/message-import-manager/crons/jobs/messaging-messages-import.cron.job.ts:74]: adding some temporary timers to troubleshoot performances

File: packages/twenty-server/src/engine/core-modules/file/controllers/file.controller.ts (hunk 1 above)
Comment (Weiko) [reply packages/twenty-server/src/engine/core-modules/file/controllers/file.controller.ts:47]: I would update the read function in local driver yes. I thought it was done though, my bad! @charlesBochet

**Review by Weiko — APPROVED** [review]
I'm surprised this has an impact, I thought the datasource was cached already? Although this would confirm my suspicion when I saw the logs a few weeks back 🤔
LGTM!

**Comment by charlesBochet** [comment]
> as an impact, I thought the datasource was cached already? Although this would confirm my suspicion when I saw the logs a fe

The datasource is cached in memory but when we deploy / scale horizontally the number of worker/servers they have to rebuild it and this cause lagging.

I have found another issue in the way we invalidate the cache, I'll add it
Regarding the file driver, I would like both drivers to have the same behavior and I don't get why the s3 is throwing exceptions while the local is not

**Pull request merged**